### PR TITLE
Do not return stale data to the caller

### PR DESCRIPTION
**Pull request by caglar10ur**

### CODE REVIEW

File: lib/apiservers/portlayer/restapi/handlers/containers_handlers.go [hunk 1]
```diff
@@ -380,7 +380,9 @@ func (handler *ContainersHandlersImpl) ContainerWaitHandler(params containers.Co
 
 	select {
 	case <-c.WaitForState(exec.StateStopped):
+		c.Refresh(context.Background())
```
Comment (hickeng) [inline lib/apiservers/portlayer/restapi/handlers/containers_handlers.go]: We already have a Refresh on this path, however it's got ordering issues:
see exec.go:145 and exec.go:156.

perhaps we should be moving the SetState call after the refresh (for success or failure)? The only one it's crucial for is stopped (exit code) but without knowing what the perf impact is I'm leaning towards state consistency.

**Comment by hickeng** [comment]
lgtm - I'll let you decide on exec or this. We'll likely have to revisit it anyway.

**Comment by mdubya66** [comment]
Lgtm

**Comment by caglar10ur** [comment]
I'll keep this as is for now and merge. We can address the ordering later

**Pull request merged**